### DISCUSSION TITLE
Parsing Nested Classes

### DISCUSSION THREAD

**Discussion by thaeta-of-diamonds** [pydantic/pydantic#10379] [Question] [answered]
I am using Pydantic for its JSON parsing and serialization for a game I am working on. Currently, I have been able to serialize all of the information I need in JSON format, but I'm unsure about how to approach parsing. I have many nested classes (For example, a Room contains a list of RoomObjects), and know I can use model_validate, but I don't know how I can do this recursively.

Here's an example of the JSON I've been able to serialize:
```
{
  "dungeon": [
    [
      [
        {
          "description": "You are in a large hole that stretches to the surface.",
          "objects": [
            {
              "id": "A1",
              "name": "North Wooden Door",
              "plaintext": "wooden door to the north",
              "definite": false,
              "visible": true,
              "dialogue": {
                "description": "The wooden door is old, and splinters as you rub your hand against it.",
                "description_enter": "You walk through the door.",
                "description_fail_enter": "The door doesn't budge.",
                "description_locked": "The door doesn't budge.",
                "description_unlocked": "The door has been unlocked."
              },
              "direction": [
                1,
                0,
                0
              ],
              "unlocked": false
            },
            {
              "id": "A3",
              "name": "Handwritten Note",
              "plaintext": "handwritten note",
              "definite": false,
              "visible": true,
              "dialogue": {
                "description": "The note says: \"Flip the switch.\""
              }
            }
          ]
        }
      ]
    ]
  ]
}
```

**Comment by Viicos** [comment, marked as answer]
> and know I can use model_validate, but I don't know how I can do this recursively.

passing the JSON to `model_validate*` will work, "recursive dicts" will be validated as models.

**Reply by thaeta-of-diamonds** [reply]
So this successfully validates the models, but they default to the root class when subclasses are involved. (Passage is a child of RoomObject, but when the dictionary is parsed, it validates those objects as RoomObjects). How can I ensure that classes are being parsed correctly?

**Reply by thaeta-of-diamonds** [reply]
I found that pickle works much better for what I'm trying to do, so I'm going to be using that instead.
